File: LiveTrackingSystem/LiveTracking.Api/Models/AppVersion.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LiveTracking.Api.Models;

public class AppVersion
{
    [Key]
    public int AppVersionId { get; set; }

    [Required]
    [MaxLength(30)]
    public string Platform { get; set; } = "Android"; // "Android", "iOS"

    public int VersionCode { get; set; }

    [Required]
    [MaxLength(50)]
    public string VersionName { get; set; } = string.Empty;

    public int MinVersionCode { get; set; } = 1;

    public bool IsForceUpdate { get; set; } = false;

    [Required]
    [MaxLength(1000)]
    public string DownloadUrl { get; set; } = string.Empty;

    [MaxLength(200)]
    public string Title { get; set; } = "New Update Available";

    public string ReleaseNotes { get; set; } = string.Empty;

    public bool IsActive { get; set; } = true;

    public int? CompanyId { get; set; }

    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;

    public Company? Company { get; set; }
}

File: LiveTrackingSystem/SmartWorkForce_Apps/app/src/main/java/com/zynexbd/livetracking/models/AppVersion.kt
package com.zynexbd.livetracking.models

import com.google.gson.annotations.SerializedName

data class AppVersionCheckResponse(
    @SerializedName("hasUpdate") val hasUpdate: Boolean = false,
    @SerializedName("isForceUpdate") val isForceUpdate: Boolean = false,
    @SerializedName("latestVersionCode") val latestVersionCode: Int = 1,
    @SerializedName("latestVersionName") val latestVersionName: String = "1.0",
    @SerializedName("downloadUrl") val downloadUrl: String = "",
    @SerializedName("title") val title: String = "",
    @SerializedName("releaseNotes") val releaseNotes: String = "",
    @SerializedName("currentVersionCode") val currentVersionCode: Int = 1
)

data class AppVersionDetails(
    @SerializedName("appVersionId") val appVersionId: Int = 0,
    @SerializedName("platform") val platform: String = "Android",
    @SerializedName("versionCode") val versionCode: Int = 1,
    @SerializedName("versionName") val versionName: String = "1.0",
    @SerializedName("minVersionCode") val minVersionCode: Int = 1,
    @SerializedName("isForceUpdate") val isForceUpdate: Boolean = false,
    @SerializedName("downloadUrl") val downloadUrl: String = "",
    @SerializedName("title") val title: String = "",
    @SerializedName("releaseNotes") val releaseNotes: String = "",
    @SerializedName("isActive") val isActive: Boolean = true,
    @SerializedName("createdAtUtc") val createdAtUtc: String = ""
)
